internal/provider/data_source_certificate.go

```diff
@@ -84,83 +84,16 @@ func (dst *certificateDataSourceType) GetSchema(_ context.Context) (tfsdk.Schema
-						MarkdownDescription: "The algorithm used to sign the certificate.",
-					},
-					"public_key_algorithm": {
-						Type:                types.StringType,
-						Computed:            true,
-						MarkdownDescription: "The key algorithm used to create the certificate.",
-					},
-					"serial_number": {
-						Type:     types.StringType,
-						Computed: true,
-						MarkdownDescription: "Number that uniquely identifies the certificate with the CA's system. " +
-							"The `format` function can be used to convert this _base 10_ number " +
-							"into other bases, such as hex.",
-					},
-					"is_ca": {
-						Type:                types.BoolType,
-						Computed:            true,
-						MarkdownDescription: "`true` if the certificate is of a CA (Certificate Authority).",
-					},
-					"version": {
-						Type:                types.Int64Type,
-						Computed:            true,
-						MarkdownDescription: "The version the certificate is in.",
-					},
-					"issuer": {
-						Type:     types.StringType,
-						Computed: true,
-						MarkdownDescription: "Who verified and signed the certificate, roughly following " +
-							"[RFC2253](https://tools.ietf.org/html/rfc2253).",
-					},
-					"subject": {
-						Type:     types.StringType,
-						Computed: true,
-						MarkdownDescription: "The entity the certificate belongs to, roughly following " +
-							"[RFC2253](https://tools.ietf.org/html/rfc2253).",
-					},
-					"not_before": {
-						Type:     types.StringType,
-						Computed: true,
-						MarkdownDescription: "The time after which the certificate is valid, as an " +
-							"[RFC3339](https://tools.ietf.org/html/rfc3339) timestamp.",
-					},
-					"not_after": {
-						Type:     types.StringType,
-						Computed: true,
-						MarkdownDescription: "The time until which the certificate is invalid, as an " +
-							"[RFC3339](https://tools.ietf.org/html/rfc3339) timestamp.",
-					},
-					"sha1_fingerprint": {
-						Type:                types.StringType,
-						Computed:            true,
-						MarkdownDescription: "The SHA1 fingerprint of the public key of the certificate.",
-					},
-					"cert_pem": {
-						Type:     types.StringType,
-						Computed: true,
-						MarkdownDescription: "Certificate data in [PEM (RFC 1421)](https://datatracker.ietf.org/doc/html/rfc1421) format. " +
-							"**NOTE**: the [underlying](https://pkg.go.dev/encoding/pem#Encode) " +
-							"[libraries](https://pkg.go.dev/golang.org/x/crypto/ssh#MarshalAuthorizedKey) that generate this " +
-							"value append a `\\n` at the end of the PEM. " +
-							"In case this disrupts your use case, we recommend using " +
-							"[`trimspace()`](https://www.terraform.io/language/functions/trimspace).",
+				Type: types.ListType{
+					ElemType: types.ObjectType{
+						AttrTypes: x509CertObjectAttrTypes(),
 					},
 				},
+				Computed: true,
+				Validators: []tfsdk.AttributeValidator{
+					listvalidator.SizeAtLeast(1),
+				},
```

Comment:
Nit: Attribute validation only applies to practitioner configuration and the `Validate*Config` RPCs, or put differently, this has no effect here since the attribute is not `Optional` or `Required` to make it configurable

```suggestion
```